src/main/java/org/reaktivity/nukleus/tls/internal/stream/ClientStreamFactory.java

```diff
@@ -352,17 +350,10 @@ private void handleBegin(
                     tlsParameters.setServerNames(asList(new SNIHostName(tlsHostname)));
                 }
 
-                String[] applicationProtocols = new String[] { applicationProtocol };
-                if (applicationProtocols.length > 0)
+                if (tlsApplicationProtocol != null && !tlsApplicationProtocol.isEmpty())
```

Comment:
We should only test for `!= null` here, since empty string is a valid protocol value.

Reply:
SSLParameters#setApplicationProtocols() doesn't allow empty string for protocol